### ISSUE TITLE
non-ASCII character in source file name breaks sphinx

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#926]
It seems like the same one in issues #227 and #451.

This is the traceback

```
# Sphinx version: 1.1.3
# Python version: 2.7.3
# Docutils version: 0.9 release
# Jinja2 version: 2.6
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/sphinx/cmdline.py", line 189, in main
    app.build(force_all, filenames)
  File "/usr/lib64/python2.7/site-packages/sphinx/application.py", line 204, in build
    self.builder.build_update()
  File "/usr/lib64/python2.7/site-packages/sphinx/builders/__init__.py", line 196, in build_update
    'out of date' &#37; len(to_build))
  File "/usr/lib64/python2.7/site-packages/sphinx/builders/__init__.py", line 216, in build
    purple, length):
  File "/usr/lib64/python2.7/site-packages/sphinx/builders/__init__.py", line 120, in status_iterator
    for item in iterable:
  File "/usr/lib64/python2.7/site-packages/sphinx/environment.py", line 613, in update_generator
    self.read_doc(docname, app=app)
  File "/usr/lib64/python2.7/site-packages/sphinx/environment.py", line 758, in read_doc
    pub.set_source(None, src_path.encode(fs_encoding))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 22: ordinal not in range(128)
```

The exception was thrown at:

```
Exception occurred:
  File "/usr/lib64/python2.7/site-packages/sphinx/environment.py", line 758, in read_doc
    pub.set_source(None, src_path.encode(fs_encoding))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 22: ordinal not in range(128)
```

This bug can be reproduced with a default project created by `sphinx-quickstart` and a source filename with non-ascii character

```
sphinx-quickstart test
... use default settings ...
cd test
touch café.rst
make html
```

The mere existence of this file (`café.rst`) will break sphinx.

Looks like in `sphinx/environment.py`, line 758 (as of rev.6e960412308f)

```
pub.set_source(None, src_path.encode(fs_encoding))

src_path}}} is actually a bytestring with encoding("utf-8" in my case) instead of what the code expected, which should be a {{{unicode}}}. 
{{{src_path
```

I tried to turn `src_path` into `unicode` with

```
pub.set_source(None, src_path.decode('utf-8').encode(fs_encoding))
```

The build process then "succeeded" with warning like

```
WARNING: a Unicode error occurred when rendering the page café. Please make sure all config values that contain non-ASCII content are Unicode strings.
```

and even if the offending file `café.rst` is listed in master doc, no output of this file will be created. So obviously this is not the way to go.

The output from `sys.getfilesystemencoding()` is `'UTF-8'`

I tested with both Python 2.7 and Python 3.2, with and without `virtualenv`.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/926
- Originally reported by: Philip Xu
- Originally created at: 2012-05-11T05:31:16.875

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-11 08:19:03+00:00_

This should have been fixed in 1.2 -- and I can't reproduce it now.

**Comment by shimizukawa** [comment]
_From Philip Xu on 2014-01-12 15:11:41+00:00_

Confirmed. Thanks Georg.
